File: cpp/Kode program/WarrantyInfo.cpp
#pragma once
#include <iostream>
#include <string>
using namespace std;

class WarrantyInfo {
private:
    string duration;           // Durasi garansi (string untuk fleksibel)
    string coverageDetails;    // Detail cakupan garansi
public:
    void setDuration(string d) { duration = d; }   // Set durasi garansi
    string getDuration() { return duration; }      // Get durasi garansi

    void setCoverageDetails(string c) { coverageDetails = c; }  // Set detail cakupan
    string getCoverageDetails() { return coverageDetails; }     // Get detail cakupan

    void printWarranty() {  // Print info garansi
        cout << "--- Warranty Info ---" << endl;
        cout << "Duration: " << duration << endl;
        cout << "Coverage Details: " << coverageDetails << endl;
    }
};

File: python/Kode program/WarrantyInfo.py
class WarrantyInfo:
    def __init__(self):
        self.duration = ""
        self.coverageDetails = ""

    def setDuration(self, d):
        self.duration = d

    def getDuration(self):
        return self.duration

    def setCoverageDetails(self, c):
        self.coverageDetails = c

    def getCoverageDetails(self):
        return self.coverageDetails

    def printWarranty(self):
        print("--- Warranty Info ---")
        print(f"Duration: {self.duration}")
        print(f"Coverage Details: {self.coverageDetails}")
